compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -553,19 +553,27 @@ object Parsers {
     def inDefScopeBraces[T](body: => T, rewriteWithColon: Boolean = false): T =
       inBracesOrIndented(body, rewriteWithColon)
 
-    /** part { `separator` part }
-     */
-    def tokenSeparated[T](separator: Int, part: () => T): List[T] = {
```

Comment:
I inlined this since it was only ever used for a comma separated list.

Reply:
That rang a bell. At the beginning of pandemic, I moved Scala 2 trailing comma handling into parser for better handling, so that is why this method still pulls its weight there.

https://github.com/scala/scala/pull/8780

I said on the PR that would forward-port it; apparently I looked and it seemed similar. But obviously I haven't done that yet. It must have slipped my mind during pandemic brain fog.

Reply:
I went ahead and wrote up something to get rid of the weird commas. I'll put it up after this PR is in.